tests/test_runner.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC Contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "uart_tests.hh"
+#include <debug.hh>
+#include <platform-uart.hh>
+#include <thread.h>
+
+using Debug = ConditionalDebug<true, "Sonata Test Runner">;
+
+[[noreturn]] void finish_running(const char *message)
+{
+	Debug::log(message);
+
+	while (true)
+	{
+		Timeout t{100};
+		thread_sleep(&t);
+	}
+}
+
+void check_result(bool result)
+{
+	if (result)
+	{
+		return;
+	}
+	finish_running("One or more tests failed");
+}
+
+[[noreturn]] void __cheri_compartment("test_runner") run_tests()
```

Comment:
Another smaller concern - I don't think we should run all tests in a single binary like this. May be convenient from a software point of view but adds more pain to the hardware debug.